### PR TITLE
[Sprint 4] Feature Logic Load ShopInfo

### PR DESCRIPTION
**Pull request by tantuan127**
Feature Get List Shop Info
Link redmine : https://dev.framgia.com/redmine/issues/116010
Task : Join Domain Shop Management
Effect: N/A

### CODE REVIEW

File: app/src/main/java/com/framgia/forder/data/model/ShopInfo.java [hunk 1]
```diff
@@ -11,42 +10,34 @@
 public class ShopInfo {
     @Expose
     @SerializedName("domain_id")
-    private int mDomainId;
+    private String mDomainId;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/data/model/ShopInfo.java]: Please keep old type of variable

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java [hunk 2]
```diff
@@ -1,87 +1,112 @@
...
 
     private final List<ShopManagement> mShopManagements;
-    private BaseRecyclerViewAdapter.OnRecyclerViewItemClickListener<Object> mItemClickListener;
 
-    public ListShopManagementAdapter(@NonNull Context context,
-            List<ShopManagement> shopManagements) {
-        super(context);
+    public ListShopManagementAdapter(List<ShopManagement> shopManagements) {
         mShopManagements = new ArrayList<>();
         if (shopManagements == null) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java]: Please remove condition, It unnecessary

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java [hunk 3]
```diff
@@ -1,87 +1,112 @@
...
-        return new ListShopManagementAdapter.ItemViewHolder(binding, mItemClickListener);
+    public int getGroupCount() {
+        return mShopManagements.size();
     }
 
     @Override
-    public void onBindViewHolder(ListShopManagementAdapter.ItemViewHolder holder, int position) {
-        holder.bind(mShopManagements.get(position));
+    public int getChildrenCount(int groupPosition) {
+        return mShopManagements.get(groupPosition).getShopInfos().size();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java]: please check ``!=null``

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java [hunk 4]
```diff
@@ -1,87 +1,112 @@
...
-        return mShopManagements.size();
+    public Object getGroup(int groupPosition) {
+        return mShopManagements.get(groupPosition);
     }
 
-    public void setItemClickListener(OnRecyclerViewItemClickListener<Object> itemClickListener) {
-        mItemClickListener = itemClickListener;
+    @Override
+    public Object getChild(int groupPosition, int childPosition) {
+        return mShopManagements.get(groupPosition).getShopInfos().get(childPosition);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java]: please check ``!=null``

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java [hunk 5]
```diff
@@ -1,87 +1,112 @@
...
-    public void updateData(List<ShopManagement> shopManagements) {
-        if (shopManagements == null) {
-            return;
-        }
-        mShopManagements.clear();
-        mShopManagements.addAll(shopManagements);
-        notifyDataSetChanged();
+    @Override
+    public long getGroupId(int groupPosition) {
+        return mShopManagements.get(groupPosition).getShop().getId();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java]: please check ``!=null``

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java [hunk 6]
```diff
@@ -1,87 +1,112 @@
...
+                .getShopInfos()
+                .get(childPosition)
+                .getDomainId());
+    }
 
-        private final ItemShopManagementBinding mBinding;
-        private final BaseRecyclerViewAdapter.OnRecyclerViewItemClickListener<Object>
-                mItemClickListener;
+    public Object getChildShopDomain(int groupPostion, int childPostion) {
+        return mShopManagements.get(groupPostion).getShopDomains().get(childPostion).getName();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java]: please check ``!=null``

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ShopManagementViewModel.java [hunk 7]
```diff
@@ -22,7 +22,7 @@
     public ShopManagementViewModel(Navigator navigator, ListShopManagementAdapter adapter) {
         mNavigator = navigator;
         mListShopManagementAdapter = adapter;
-        mListShopManagementAdapter.setItemClickListener(this);
+        //        mListShopManagementAdapter.setItemClickListener(this);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ShopManagementViewModel.java]: remove this line and implement if unused

File: app/src/main/java/com/framgia/forder/data/model/ShopInfo.java [hunk 8]
```diff
@@ -17,25 +16,17 @@
     private String mDomainName;
     @Expose
     @SerializedName("number_of_users")
-    private int mNumberUser;
+    private String mNumberUser;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/data/model/ShopInfo.java]: Please keep old type of variable

File: app/src/main/java/com/framgia/forder/data/model/ShopInfo.java [hunk 9]
```diff
@@ -17,25 +16,17 @@
...
     @Expose
     @SerializedName("number_of_shops")
-    private int mNumberShop;
+    private String mNumberShop;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/data/model/ShopInfo.java]: Please keep old type of variable

File: app/src/main/java/com/framgia/forder/data/model/ShopInfo.java [hunk 10]
```diff
@@ -17,25 +16,17 @@
...
     @Expose
     @SerializedName("number_of_products")
-    private int mNumberProduct;
+    private String mNumberProduct;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/data/model/ShopInfo.java]: Please keep old type of variable

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java [hunk 11]
```diff
@@ -1,87 +1,124 @@
...
-        extends BaseRecyclerViewAdapter<ListShopManagementAdapter.ItemViewHolder> {
+public class ListShopManagementAdapter extends BaseExpandableListAdapter {
 
     private final List<ShopManagement> mShopManagements;
-    private BaseRecyclerViewAdapter.OnRecyclerViewItemClickListener<Object> mItemClickListener;
 
-    public ListShopManagementAdapter(@NonNull Context context,
-            List<ShopManagement> shopManagements) {
-        super(context);
+    public ListShopManagementAdapter(List<ShopManagement> shopManagements) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java]: should remove params ``List<ShopManagement>`` from this method

Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java]: Sorry It my mistake in sample, Please fix

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java [hunk 12]
```diff
@@ -1,87 +1,124 @@
...
-                mItemClickListener;
-
-        ItemViewHolder(ItemShopManagementBinding binding,
-                BaseRecyclerViewAdapter.OnRecyclerViewItemClickListener<Object> listener) {
-            super(binding.getRoot());
-            mBinding = binding;
-            mItemClickListener = listener;
+    @Override
+    public long getChildId(int groupPosition, int childPosition) {
+        if (mShopManagements.get(groupPosition).getShopInfos().get(childPosition) != null) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java]: Please change to
```java
if(mShopManagements.get(groupPosition)!=null && mShopManagements.get(groupPosition).getShopInfos()!=null && mShopManagements.get(groupPosition).getShopInfos().get(childPosition!=null){
// do something
}
```

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java [hunk 13]
```diff
@@ -1,87 +1,124 @@
...
         }
+        return 0;
+    }
 
-        void bind(ShopManagement shopManagements) {
-            mBinding.setViewModel(
-                    new ItemShopManagementViewModel(shopManagements, mItemClickListener));
-            mBinding.executePendingBindings();
+    public Object getChildShopDomain(int groupPostion, int childPostion) {
+        if (mShopManagements.get(groupPostion).getShopDomains().get(childPostion) != null) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java]: like above comment!

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java (hunk 11 above)
Comment (tantuan127) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java]: Added

File: app/src/main/res/values/strings.xml [hunk 14]
```diff
@@ -95,9 +94,10 @@
     <string name="reject">Reject</string>
     <string name="select_all">Select all</string>
     <string name="see_more">See More..</string>
-    <string name="user">User :</string>
-    <string name="products">Product :</string>
-    <string name="shops">Shop :</string>
+    <string name="user">User : %d</string>
```
Comment (daolq3012) [inline app/src/main/res/values/strings.xml]: rename to ``user_with_count``

File: app/src/main/res/values/strings.xml [hunk 15]
```diff
@@ -95,9 +94,10 @@
...
+    <string name="products">Product : %d</string>
```
Comment (daolq3012) [inline app/src/main/res/values/strings.xml]: like above

File: app/src/main/res/values/strings.xml [hunk 16]
```diff
@@ -95,9 +94,10 @@
...
+    <string name="shops">Shop : %d</string>
```
Comment (daolq3012) [inline app/src/main/res/values/strings.xml]: like above

File: app/src/main/res/values/strings.xml [hunk 17]
```diff
@@ -95,9 +94,10 @@
...
+    <string name="statusDomain">Status : %s</string>
```
Comment (daolq3012) [inline app/src/main/res/values/strings.xml]: rename to ``status_domain``

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java [hunk 18]
```diff
@@ -3,85 +3,133 @@
...
+        return binding.getRoot();
+    }
+
+    @Override
+    public View getChildView(int groupPosition, int childPosition, boolean isLastChild,
+            View convertView, ViewGroup parent) {
+        ItemJoinDomainBinding binding =
+                DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()),
+                        R.layout.item_join_domain, parent, false);
+        ItemShopInfoViewModel viewModel = new ItemShopInfoViewModel(mContext,
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ListShopManagementAdapter.java]: change param ``mContext`` to ``mContext.getApplicationContext()``

**Pull request merged**